### PR TITLE
Audit Engine Plus

### LINKED ISSUE #383: Investigate ballots to audit count on county dashboard
The county dashboard ballots to audit count was double what it should have been in an experiment @kiniry and I ran tonight - it seemed to be counting ballots from multiple counties; investigate this to see what might be going on there.

### PR DESCRIPTION
**Pull request by dmzimmerman**
A complete rearchitecture of the audit engine, giving per-county-per-contest results and the ability to set individual parameters on that basis, as well as lazily creating the list of ballots to audit based on the requirements of audit rounds. 

Also closes #383

### CODE REVIEW

File: server/eclipse-project/pom.xml [hunk 1]
```diff
@@ -4,7 +4,7 @@
 	<modelVersion>4.0.0</modelVersion>
 	<groupId>us.freeandfair.production</groupId>
 	<artifactId>colorado_rla</artifactId>
-	<version>0.9.1</version>
+	<version>0.10.0</version>
```
Comment (kiniry) [inline server/eclipse-project/pom.xml]: Double digits!

File: server/eclipse-project/src/main/java/us/freeandfair/corla/controller/ComparisonAuditController.java [hunk 2]
```diff
@@ -0,0 +1,350 @@
...
+  private ComparisonAuditController() {
+    // empty
+  }
+  
+  /**
+   * Compute the ballot (cards) for audit, for a particular county dashboard, 
+   * random seed, and index range.
+   * 
+   * @param the_cdb The county dashboard.
+   * @param the_seed The random seed.
```
Comment (kiniry) [inline server/eclipse-project/src/main/java/us/freeandfair/corla/controller/ComparisonAuditController.java]: Too many @params in the Javadocs.

File: server/eclipse-project/src/main/java/us/freeandfair/corla/controller/ComparisonAuditController.java [hunk 3]
```diff
@@ -0,0 +1,350 @@
...
+    the_dashboard.setEstimatedBallotsToAudit(computeEstimatedBallotsToAudit(the_dashboard));
+    
+    return result;
+  }
+  
+  /**
+   * Audits a CVR/ACVR pair by adding it to all the audits in progress.
+   * This also updates the local audit counters, as appropriate.
+   * 
+   * @param the_cvr_under_audit The CVR under audit.
```
Comment (kiniry) [inline server/eclipse-project/src/main/java/us/freeandfair/corla/controller/ComparisonAuditController.java]: Missing @param for the_dashboard.

File: server/eclipse-project/src/main/java/us/freeandfair/corla/crypto/PseudoRandomNumberGenerator.java [hunk 4]
```diff
@@ -126,22 +126,22 @@ public PseudoRandomNumberGenerator(final String the_seed,
...
    * @param the_to the "index" of the final random number to give
    * 
-   * @return A list containing the_from - the_to + 1 random numbers
+   * @return A list containing the_to - the_from + 1 random numbers
    */
   //@ requires the_from <= the_to;
   // @todo kiniry Refine this specification to include public model fields.
-  // requires my_with_replacement;
-  // requires the_to <= my_maximum_index;
+  // requires my_with_replacement || the_to <= my_maximum_index;
```
Comment (kiniry) [inline server/eclipse-project/src/main/java/us/freeandfair/corla/crypto/PseudoRandomNumberGenerator.java]: Why is this disjunction?

File: server/eclipse-project/src/main/java/us/freeandfair/corla/controller/ComparisonAuditController.java (hunk 2 above)
Comment (dmzimmerman) [reply server/eclipse-project/src/main/java/us/freeandfair/corla/controller/ComparisonAuditController.java]: Fixed in next commit.

File: server/eclipse-project/src/main/java/us/freeandfair/corla/controller/ComparisonAuditController.java (hunk 3 above)
Comment (dmzimmerman) [reply server/eclipse-project/src/main/java/us/freeandfair/corla/controller/ComparisonAuditController.java]: Fixed in next commit.

File: server/eclipse-project/src/main/java/us/freeandfair/corla/crypto/PseudoRandomNumberGenerator.java (hunk 4 above)
Comment (dmzimmerman) [reply server/eclipse-project/src/main/java/us/freeandfair/corla/crypto/PseudoRandomNumberGenerator.java]: If we're sampling with replacement, we don't need to worry about how many numbers we're asking for; if we're not, we do.

**Pull request merged**